slowapi/extension.py

```diff
@@ -354,6 +354,10 @@ def _inject_headers(
         self, response: Response, current_limit: Tuple[RateLimitItem, List[str]]
     ) -> Response:
         if self.enabled and self._headers_enabled and current_limit is not None:
+            if not isinstance(response, Response):
```

Comment:
What do you think of using `assert isinstance(response, Response), "your error message here"` instead? It allows a tiny optimisation when running python in non-debug mode (and then I'd argue that the error is on the user if they still have invalid code).
Edit: I just noticed that the original code uses the pattern you used as well. Maybe I should update that part as well. What do you think?

Reply:
Yes, I re-used the pattern you had for the `Request`. I don't have a strong opinion for each. I had to migrate a few endpoints and add the request/response, during this process I forgot a few so having the explicit exception helped a bit. 

For FastAPI, I am actually thinking it would be better to wrap the functionality as a [dependency](https://fastapi.tiangolo.com/tutorial/dependencies/) rather than a decorator and this would drop the need to have these checks and the requirement to add `request` and `response` to the endpoint.
I don't have enough knowledge with Starlette to see if something similar could help.

Reply:
That's a neat idea! I haven't seen anything similar for Starlette (Dependency injection), so I'd want to check a bit more to make sure this does not require adding lots more code just to do the same thing in a different way. If it's just a different entry point into the same functions, it would be pretty cool! I've opened #23 to track this specifically outside of the current PR.